lib/acme/client.rb

```diff
@@ -47,6 +49,12 @@ def new_certificate(csr)
     ::Acme::Certificate.new(OpenSSL::X509::Certificate.new(response.body), fetch_chain(response), csr)
   end
 
+  def new_self_signed_certificate domain
+    cert = ::Acme::Certificate.new(nil, nil, nil)
+    cert.create_self_signed_cert([domain])
```

Comment:
`tls-sni-01` has the constraint of the name having to be in a cert in a SAN. This is rather untypical for a certificate only covering a single domain, and thus `Acme::CertificateRequest` for example generates only the `CN` but no SAN in the single domain case. Thus I think we should keep the logic to generate a self signed `OpenSSL::X509::Certificate` inside the challenge class and pass it to `Acme::Certificate`.